homeassistant/config.py

```diff
@@ -1025,21 +1045,175 @@ async def merge_packages_config(
+            "config_validator_unknown_err": (
+                f"Unknown error calling {domain} config validator",
+                True,
+            ),
+            "config_schema_unknown_err": (
+                f"Unknown error calling {domain} CONFIG_SCHEMA",
+                True,
+            ),
+            "platform_validator_unknown_err": (
+                f"Unknown error validating {platform_name} platform config with {domain} "
+                "component platform schema",
+                True,
+            ),
+            "platform_component_load_err": (f"Platform error: {domain} - {ex}", False),
+            "platform_component_load_exc": (f"Platform error: {domain} - {ex}", True),
+            "platform_schema_validator_err": (
+                f"Unknown error validating config for {platform_name} platform "
+                f"for {domain} component with PLATFORM_SCHEMA",
+                True,
+            ),
+        }
+        return log_message_mapping.get(translation_key, (None, False))
+
+    integration_config_info = await async_process_component_config(
+        hass, config, integration
+    )
+    processed_config = integration_config_info.config
+    config_exception_info = integration_config_info.exception_info_list
+    if not config_exception_info:
+        return processed_config
+
+    platform_exception: ConfigExceptionInfo
+    config_error_messages: list[
+        tuple[str, ConfigExceptionInfo, str, str, int | str]
+    ] = []
+    general_error_messages: list[tuple[str, ConfigExceptionInfo]] = []
+    domain = integration.domain
+    for platform_exception in config_exception_info:
+        link = platform_exception.integration_link or integration.documentation
+        ex = platform_exception.exception
+        p_name = platform_exception.platform_name
+        p_config = platform_exception.config
+        config_file: str = "?"
+        line: int | str = "?"
+        log_message, show_stack_trace = _get_log_message_and_stack_print_pref(
+            platform_exception.translation_key, ex, domain, p_name
+        )
+        if log_message is None:
+            # No pre defined log_message is set, so we generate a message
+            if isinstance(ex, vol.Invalid):
+                log_message = format_schema_error(hass, ex, p_name, p_config, link)
+                if annotation := find_annotation(p_config, ex.path):
+                    config_file, line = annotation
+            else:
+                if TYPE_CHECKING:
+                    assert isinstance(ex, HomeAssistantError)
+                log_message = format_homeassistant_error(
+                    hass, ex, p_name, p_config, link
+                )
+                if annotation := find_annotation(p_config, [p_name]):
+                    config_file, line = annotation
+                show_stack_trace = True
+            config_error_messages.append(
+                (domain, platform_exception, log_message, config_file, line)
+            )
```

Comment:
Please make this a dataclass

Reply:
The lists were removed so a dataclass is not needed afterall